flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/datastream/DataStream.java

```diff
@@ -617,7 +617,7 @@ protected ExecutionConfig getExecutionConfig() {
 	 */
 	public <R> SingleOutputStreamOperator<R, ?> fold(R initialValue, FoldFunction<OUT, R> folder) {
 		TypeInformation<R> outType = TypeExtractor.getFoldReturnTypes(clean(folder), getType(),
-				Utils.getCallLocationName(), false);
+				Utils.getCallLocationName(), true);
```

Comment:
I set allowMissing to true as it was set at flatMap. Leaving it on false caused the call fold in the Scala DataStream fail.